packages/nextjs/components/BetManager.tsx

```diff
@@ -12,13 +95,28 @@ function setBetManagerActiveTab(activeTab: BetManagerTabEnum) {
   localStorage.setItem("betManagerActiveTab", activeTab);
 }
 
+/**
+ * Converts a string or string array to a numeric value.
+ * - If `s` is a string, the function parses it as a hexadecimal character.
+ * - If `s` is a string array, the function joins the elements and parses the result as a binary number.
+ * @param s - A single string or an array of strings.
+ * @returns The parsed numeric value.
+ */
+function getHashPickValue(s: string | string[]): number {
+  if (typeof s === "string") {
+    return parseInt(s, 16);
+  }
+  return parseInt(s.join(""), 2);
+}
+
 const BetManager = ({ writePlaceBet }: { writePlaceBet: any }) => {
   const [activeTab, setActiveTab] = useState<BetManagerTabEnum>(() => {
     return (localStorage.getItem("betManagerActiveTab") as BetManagerTabEnum) || BetManagerTabEnum.PLACE_BET;
   });
+  const ethPrice = 3800;
```

Comment:
hard coded for now but we should consume a ws stream

Reply:
and if we do that, we can wire it up to the setEthPrice hook in the betAMount reducer

Reply:
Check out: useInitializeNativeCurrencyPrice
It looks like this fetches the current price from Uniswap.